File: icamera/src/main/java/me/shouheng/icamera/enums/CameraType.java
package me.shouheng.icamera.enums;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import static me.shouheng.icamera.enums.CameraType.*;

/**
 * Camera1 or camera2
 *
 * @author <a href="mailto:shouheng2015@gmail.com">WngShhng</a>
 * @version 2019-12-28 16:39
 */
@IntDef(value = {TYPE_CAMERA1, TYPE_CAMERA2})
@Retention(value = RetentionPolicy.SOURCE)
public @interface CameraType {

    /** Camera1 */
    int TYPE_CAMERA1        = 0x0100;

    /** Camera2 */
    int TYPE_CAMERA2        = 0x0200;
}

File: icamera/src/main/java/me/shouheng/icamera/enums/CameraType.kt
package me.shouheng.icamera.enums

import androidx.annotation.IntDef

/**
 * Camera1 or camera2
 *
 * @author [WngShhng](mailto:shouheng2015@gmail.com)
 * @version 2019-12-28 16:39
 */
@IntDef(value = [CameraType.TYPE_CAMERA1, CameraType.TYPE_CAMERA2])
@kotlin.annotation.Retention(AnnotationRetention.SOURCE)
annotation class CameraType {
    companion object {
        /** Camera1  */
        const val TYPE_CAMERA1 = 0x0100
        /** Camera2  */
        const val TYPE_CAMERA2 = 0x0200
    }
}